fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -817,8 +796,36 @@ public int getAliveComputeNodeNumber() {
     }
 
     public ComputeNode getComputeNodeWithBePort(String host, int bePort) {
-        for (ComputeNode computeNode : idToComputeNodeRef.values()) {
-            if (computeNode.getHost().equals(host) && computeNode.getBePort() == bePort) {
+        return getComputeNodeWithBePortCommon(host, bePort, idToComputeNodeRef);
+    }
+
+    private ComputeNode getComputeNodeWithBePortCommon(String host, int bePort, Map<Long, ? extends ComputeNode> nodeRef) {
```

Comment:
```suggestion
    private <T extends ComputeNode> T getComputeNodeWithBePortCommon(String host, int bePort, Map<Long, T> nodeRef) {
```

Reply:
Avoid downward casting from ComputeNode to Backend

Reply:
> Avoid downward casting from ComputeNode to Backend

Seems like still need to cast to `T` when return from `getComputeNodeWithBePortCommon ` . So I will just leave it as it is now

Reply:
T is the element type, no need to cast, the.for loop should use T instead of computeNode